### PR TITLE
Added ToggleStatePropertyChanged UIA event for ToolStripButton.

### LINKED ISSUE #9137: Screen readers don't announce checked state change of ToolStripButton
### .NET version

.NET 8

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

When a `ToolStripButton` with `CheckOnClick = true` is pressed (via keyboard or mouse), screen readers don't announce it. For example, this is Narrator output when focusing on a checkable `ToolStripButton` and pressing Space key: 

> toolStrip1, toolStripButton1 check box unchecked, 
> Space

ToolStripButton becomes checked, but it's not announced. On the other hand, activating a `CheckBox` is announced:

> checkBox1 check box unchecked, 
> Space
> checkBox1 check box checked, 

Other checkable controls (DataGridView cell, CheckedListBox row, TreeView node) are also announced similar to CheckBox.

### Steps to reproduce

1. Add a ToolStrip to the form
2. Add ToolStripButton to the ToolStrip, set `CheckOnClick = true`
3. Start app, focus on the ToolStripButton
4. Press space/enter or click mouse

### PR DESCRIPTION
**Pull request by v-elnovikova**
Fixes #9137

## Proposed changes

- Fire ToggleStatePropertyChanged UIA event when ToolStripButton's `CheckState` changes.

## Customer Impact

- Screen readers announce checked state change of ToolStripButton

## Regression? 

- No

## Risk

- Minimal

### Before

https://github.com/dotnet/winforms/assets/113603457/525781bb-2a75-4fe2-8fe8-ab863141f90c

Narrator output:
> ToolStripTests window, toolStrip1 tool bar, Checkable button (unchecked) check box unchecked,

Notice that nothing is announced when button is checked or unchecked.

### After

https://github.com/dotnet/winforms/assets/113603457/c986f680-aecb-4754-9de5-50f9b2f4e2c2
Narrator output:
> ToolStripTests window, toolStrip1 tool bar, Checkable button (unchecked) check box unchecked,
> checked
> unchecked
> checked
> unchecked

## Test environment(s) 

- ,NET  8.0.0-preview.4.23259.5

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9142)

### CODE REVIEW

**Review by Tanya-Solyanik — APPROVED** [review]
Looks good! Thank you for the fast turn around.

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripButton.ToolStripButtonAccessibleObject.cs [hunk 1]
```diff
@@ -76,8 +76,21 @@ internal override void Toggle()
...
+        {
+            RaiseAutomationPropertyChangedEvent(
+                UiaCore.UIA.ToggleToggleStatePropertyId,
+                CheckedStateToToggleState(oldValue),
+                CheckedStateToToggleState(newValue));
+
+            RaiseAutomationEvent(UiaCore.UIA.AutomationPropertyChangedEventId);
+        }
+
+        private static UiaCore.ToggleState CheckedStateToToggleState(CheckState checkState)
```
Comment (dmitrii-drobotov) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripButton.ToolStripButtonAccessibleObject.cs:92]: Nit:
```suggestion
        private static UiaCore.ToggleState CheckStateToToggleState(CheckState checkState)
```

**Review by dmitrii-drobotov — APPROVED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripButton.ToolStripButtonAccessibleObject.cs [hunk 2]
```diff
@@ -76,8 +76,21 @@ internal override void Toggle()
...
+            => CheckStateToToggleState(_ownerItem.CheckState);
+
+        internal void OnCheckStateChanged(CheckState oldValue, CheckState newValue)
+        {
+            RaiseAutomationPropertyChangedEvent(
+                UiaCore.UIA.ToggleToggleStatePropertyId,
+                CheckStateToToggleState(oldValue),
+                CheckStateToToggleState(newValue));
+
+            RaiseAutomationEvent(UiaCore.UIA.AutomationPropertyChangedEventId);
```
Comment (dmitrii-drobotov) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripButton.ToolStripButtonAccessibleObject.cs:89]: @v-elnovikova - is this method call needed? I would assume calling only `RaiseAutomationPropertyChangedEvent` is enough.

Comment (v-elnovikova) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripButton.ToolStripButtonAccessibleObject.cs]: Right, second method does the same thing, there's no need to call it.

**Review by dmitrii-drobotov — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]

**Pull request merged**